private/rules/maven_install.bzl

```diff
@@ -123,6 +123,7 @@ def maven_install(
             # created from the maven_install.json file in the coursier_fetch
             # invocation after this.
             name = name if maven_install_json == None else "unpinned_" + name,
+            user_provided_name = name,
```

Comment:
Instead of adding another parameter, why not derive this from `pinned_repo_name`, and if it's `None` or not an attr in the case of `pinned_coursier_fetch`, use `name`?

Reply:
Inside a repo rule,`repository_ctx.attr.name` is the fully-qualified repo name; it is no longer the shortname passed when declaring the repo. Using that value yields the following test failure:

```
INFO: From Testing //tests/unit/manifest_stamp:diff_manifest_test:
==================== Test output for //tests/unit/manifest_stamp:diff_manifest_test:
2c2
< Target-Label: @manifest_stamp_testing//:com_google_guava_guava
---
> Target-Label: @_main~maven~manifest_stamp_testing//:com_google_guava_guava
FAIL: files "tests/unit/manifest_stamp/guava_MANIFEST.MF.golden.unix" and "tests/unit/manifest_stamp/guava_MANIFEST.MF" differ. 
```

IIUC, the preferred way to deal with this is to pass a second string-valued attr, that gets the same value as `name`, at the declaration site; since this attr is user-declared, it doesn't have any platform magic associated with it - it keeps the same value into the repo rule.

So, the way I see it, we have a need for two user-declared attrs: the shortname of this repo, and the shortname of the assocaited pinned repo (in the case of unpinned repos). I could be wrong, and maybe there's some way to get by without yet another attr here, but I couldn't work out how to do it - the magic associated with the `name` attr interferes with what we're trying to do.

Reply:
Neat. Should we go with that change instead? Happy to close this if we do; I only care about the result.